internal/plugin/slo/contrib/error_budget_exhausted_alert_v1/plugin.go

```diff
@@ -0,0 +1,115 @@
+package plugin
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/model/rulefmt"
+
+	pluginslov1 "github.com/slok/sloth/pkg/prometheus/plugin/slo/v1"
+)
+
+const (
+	PluginVersion = "prometheus/slo/v1"
+	PluginID      = "sloth.dev/contrib/error_budget_exhausted_alert/v1"
+)
+
+type Config struct {
+	Threshold      float64           `json:"threshold"`                 // default 0, fully exhausted
+	For            string            `json:"for"`                       // default 5m
+	ForDuration    model.Duration    `json:"-"`                         // parsed duration, not serialized
+	Annotations    map[string]string `json:"annotations,omitempty"`     // default empty, additional annotations to add to the alert
+	AlertName      string            `json:"alert_name,omitempty"`      // default "ErrorBudgetExhausted"
+	SelectorLabels map[string]string `json:"selector_labels,omitempty"` // default empty, additional labels to determine what should alert
+	AlertLabels    map[string]string `json:"alert_labels,omitempty"`    // default empty, additional labels to add to the alert
+}
+
+func NewPlugin(configData json.RawMessage, _ pluginslov1.AppUtils) (pluginslov1.Plugin, error) {
+	cfg := Config{
+		Threshold:      0,
+		AlertLabels:    map[string]string{},
+		Annotations:    map[string]string{},
+		SelectorLabels: map[string]string{},
+	}
+	if err := json.Unmarshal(configData, &cfg); err != nil {
+		return nil, fmt.Errorf("invalid plugin config: %w", err)
+	}
+
+	// Set defaults if not empty.
+	if cfg.For == "" {
+		cfg.For = "5m"
+	}
+	if cfg.AlertName == "" {
+		cfg.AlertName = "ErrorBudgetExhausted"
+	}
+
+	// Validate for duration
+	duration, err := model.ParseDuration(cfg.For)
+	if err != nil {
+		return nil, fmt.Errorf("invalid 'for' duration in plugin config: %w", err)
+	}
+	cfg.ForDuration = duration
+
+	return plugin{config: cfg}, nil
+}
+
+type plugin struct {
+	config Config
+}
+
+// labelMatcher takes a map of labels and returns a string for PromQL inclusion.
+func labelMatcher(labels map[string]string) string {
```

Comment:
You may use a prom [`LabelSet` instead](https://pkg.go.dev/github.com/prometheus/common@v0.66.1/model#LabelSet.String).  Example:

```go
x := map[string]string{"x1": "y2", "x2": "y2"}
ls := prommodel.LabelSet{}
for k, v := range x {
    ls[prommodel.LabelName(k)] = prommodel.LabelValue(v)
}
ls.String()
```